modules/standard/Path.chpl

```diff
@@ -457,4 +457,78 @@ proc file.realPath(): string throws {
      return ("", name);
    }
  }
+
+/* Returns the argument.  If the argument begins with `~` or `~<user>`, replace
+   that substring with that user's home directory, otherwise returns the
+   argument unchanged.
+
+   :arg path: a file or directory, which may or may not include `~` or `~<user>`
+   :type path: `string`
+
+   :return: the argument, replacing any `~` or `~<user>` with the user's home
+                directory
+   :rtype: `string`
+*/
+proc expandUser(path: string) : string
+  {
+    var i: int = 1;
+    var j: int = 1;
+    var k: int = 1;
+    var lenz: int;
+    var lenp: int = path.length;;
+    var count: int = 0;
+    var x: c_string ;
+    var y: c_string;
+       
+    //If the argument begins with `~` or `~<user>`, it replaces that substring with that user's home directory
+    if path[1] == '~' 
+    {
+      //checks the present directory by PWD
+      x = "PWD";
+      sys_getenv(x,y);
+      var  z = y:string;
+      lenz = z.length;
```

Comment:
Okay, I figured out why I was confused by the PWD portion.  This code assumes that the directory where the program was run will be somewhere in the user's home directory hierarchy.  However, this isn't always the case - sometimes the user will be working in a mount that is separate from their home directory and they'll want to get the same answer regardless.

For example, say we have a file system defined like this:
```
- /
  - homes
     - Lydia
     - someoneElse
  - placeWhereIWork
     - myTestDir
  - placeWhereSomeoneElseWorks
     -theirTestDir
```
I'll want the result of this call to return the same thing regardless of if I'm executing as me in `/homes/Lydia` or `/placeWhereIWork/myTestDir`, but the code as it is implemented today would give different answers depending on where it was run.

This will take a bit more effort to implement.  I recommend taking a look at how python implements this function - you may need to add some explicit runtime support, similar to what [the chmod implementation](https://github.com/chapel-lang/chapel/pull/618) has done, for instance, or perhaps something more involved.  If you do end up using aspects of the python implementation, be sure to cite your work, similar to [this](https://github.com/chapel-lang/chapel/blob/master/modules/standard/FileSystem.chpl#L314).